app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CategoriesController < ApplicationController
+  before_action :set_category, only: %i[show edit update destroy]
+
+  def index
+    @categories = Category.all
+  end
+
+  def show
+    @category = Category.find(params[:id])
+    @group_details = Expense.joins(:categorizations, :categories)
+      .where('expenses.id = categorizations.expense_id')
+      .where('categories.id = categorizations.category_id')
+      .where('categories.id = ?', params[:id]).order('expenses.created_at DESC')
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @author = current_user
+    @category = Category.new(category_params)
+    @category.author = @author
+
+    respond_to do |format|
+      if @category.save
+        format.html { redirect_to @category, notice: 'Category was successfully created.' }
+        format.json { render :show, status: :created, location: @category }
```

Comment:
###  "Add a new category or Group" page

- [ ] According to the project requirements, When the user clicks (or taps) the "Save" button to create the new category, it must take the user to the home or categories Page on success and not the `category/:id` page. Kindly correct this as well. ✍🏻